Update design and use SkyView branding

drakeapps opened the pull request:
* Update the styling of page
* Use SkyView name instead of fa-dump1090
* Show aircraft altitude colors in aircraft paths
* Show small details pane if hovering over plane
* Add altitude chart
* Update icons
* Add back in current date/time display to page

mutability commented on public_html/planeObject.js:
```diff
@@ -300,30 +296,80 @@ PlaneObject.prototype.getMarkerColor = function() {
         var outline = (this.position_from_mlat ? OutlineMlatColor : OutlineADSBColor);
         var baseMarker = getBaseMarker(this.category, this.icaotype, this.typeDescription, this.wtc);
-        var weight = ((this.selected && !SelectedAllPlanes ? 2 : 1) / baseMarker.scale / scaleFactor).toFixed(1);
+        //var weight = ((this.selected && !SelectedAllPlanes ? 2 : 1) / baseMarker.scale / scaleFactor).toFixed(1);
```
Any reason to keep these lines?

mutability commented on public_html/planeObject.js:
```diff
@@ -300,30 +296,80 @@ PlaneObject.prototype.getMarkerColor = function() {
+                        scale: 1.2 * scaleFactor,
                         imgSize: baseMarker.size,
-                        src: svgPathToURI(baseMarker.path, baseMarker.size, outline, weight, col, transparentBorderWidth),
+                        src: svgPathToURI(baseMarker.svg, baseMarker.size, outline, '', col, ''),
```
Let's get rid of the redundant args here.

mutability commented on public_html/planeObject.js:
```diff
@@ -300,30 +296,80 @@ PlaneObject.prototype.getMarkerColor = function() {
+        //var weight = ((this.selected && !SelectedAllPlanes ? 2 : 1) / baseMarker.scale / scaleFactor).toFixed(1);
         var rotation = (this.track === null ? 0 : this.track);
-        var transparentBorderWidth = (32 / baseMarker.scale / scaleFactor).toFixed(1);
+        //var transparentBorderWidth = (32 / baseMarker.scale / scaleFactor).toFixed(1);
```
Are the new icons still easy to hit? The transparent border thing was because I kept missing the old icons when they were scaled down.

drakeapps replied:
I don't really have a problem clicking them, but I guess that varies to the user. It makes it equal to how clickable icons are on FA so making that consistent. I did find it annoying trying to click planes that are nearly on top of each with the transparent border. But it can be added back in if it seems like it should be

mutability commented on public_html/script.js:
```diff
@@ -1391,3 +1599,57 @@ function getAirframesModeSLink(code) {
+function flightFeederCheck() {
+	$.ajax('/status.json', {
+		success: function(data) {
+			if (typeof data.serial !== 'undefined') {
```
I updated the FF status.json in 7.6.0; there is now a `"type": "flightfeeder"` key that is probably better to check than looking for the serial number (uncommissioned FFs won't have a serial number)

mutability commented on public_html/style.css:
```diff
@@ -199,6 +222,489 @@ select.error, textarea.error, input.error {
+}
+
+#altitude_chart_button {
+    background-image: url('images/alt_legend_feet.svg');
```
These svgs have a fixed set of colors so they are going to be out of sync with the endpoints used in the code.

drakeapps replied:
Do you mean that if the plane is selected or stale it'll be a slightly different color and won't match up? Or something else that I'm missing?

mutability replied:
If the user changes the color configuration, the aircraft will change to reflect their modifications but the legends won't.

mutability replied:
If generating the gradient on the fly turns out to be tricky, another alternative is to disable the legend if the config.js settings don't match the defaults

drakeapps replied:
Changed so the chart will be hidden if using custom colors. 

Generating the colors for a custom chart is pretty straight forward, but making it look good and consistent is the issue.

mutability commented on public_html/script.js:
```diff
@@ -1308,6 +1473,14 @@ function onDisplayUnitsChanged(e) {
     });
 }
 
+function setAltitudeLegend(units) {
```
Is there a way to hide the legend entirely? It takes up a lot of screen real estate

drakeapps replied:
Added an option to toggle it on and off.

mutability commented:
Also can you change the base branch to dev as that's where it will be merged?
(I can do it, but it might cause random fallout so it's probably best if you do it)

Romeo-Golf commented:
Out of interest which of the many different things called skyview is this being named after? ;)

Is it this ADS-B receiver:

http://www.dynonavionics.com/skyview-system.php

Or the NASA site:

http://skyview.gsfc.nasa.gov/current/cgi/titlepage.pl

Or the iPhone app:

https://www.terminaleleven.com/skyview/iphone/

Or weather station:

http://www.skyview.co.uk

mutability commented:
None of the above.

mutability commented:
Datablock on hover is slow to appear (maybe 750ms hover time?)

The hover datablock righthand margin needs some padding:

![screenshot from 2017-03-16 18-01-43](https://cloud.githubusercontent.com/assets/3055062/24011487/d78ebb82-0a72-11e7-9245-e03d1b467885.png)

drakeapps commented:
* I looked over the follow plane functionality. Double clicking on the table works again
* I get the discontinuity on the current version also
![screenshot 2017-03-18 13 27 29](https://cloud.githubusercontent.com/assets/3067165/24088516/50a78750-0cf9-11e7-8f05-f4dce61fd4f1.png)
* Yeah, highlighting the selected aircraft probably should be better. 
* I didn't touch the size of the rings, just added the label. But can increase it
* Not 100% sure what you mean. Should the table header hover/scroll with it? It doesn't look like it was before. 

* Added padding to the hover box

mutability commented:
OK I think if we can get the selected-aircraft-highlight restored in some form then this should be good to merge, the other stuff is enhancements or existing bugs that we can look at later.

drakeapps commented:
Added a 1px black stroke to selected aircraft

mutability commented:
Need to add the outline only when a single aircraft is selected, not when select-all is used (see the original code)

drakeapps commented:
Done
